format-common/src/main/java/io/cdap/plugin/format/plugin/FileSinkProperties.java

```diff
@@ -72,6 +73,14 @@ default void validate(FailureCollector collector, Map<String, String> arguments)
    */
   String getPath();
 
+  /**
+   * Get the path to write to, with provided stage context, this can be used when the path is determined by some
+   * runtime arguments. The context is null at configure time
+   */
+  default String getPath(@Nullable StageContext context) {
```

Comment:
it's odd to me that the context is nullable. Where are we calling with null and can we just call getPath() in those situations instead?

Reply:
I thought the method will be called in configurePipeline, but it is actually only called in prepareRun, so context should never be null, removed nullable here